### PR TITLE
use CYTHON_PEP489_MULTI_PHASE_INIT on PyPy

### LINKED ISSUE #5413: [BUG] NumPy emits ImportWarning when importing numpy.random on PyPy3.9 wh
### Describe the bug

```
$ pypy3.9 -Werror -c "import numpy"
numpy/__init__.py:152: in <module>
    from . import random
numpy/random/__init__.py:180: in <module>
    from . import _pickle
numpy/random/_pickle.py:1: in <module>
    from .mtrand import RandomState
numpy/random/mtrand.pyx:19: in init numpy.random.mtrand
    ???
E   ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
```

The code imports another cython c-extension module from inside a cython c-extension module.

### Code to reproduce the behaviour:

The exception points to this

```cython
from ._mt19937 import MT19937 as _MT19937
```

### Expected behaviour

Without `-Werror` the import works, so it is "just" the warning that is problematic.

### OS

Ubuntu 22.04

### Python version

pypy3.10 (pre-release), pypy3.9

### Cython version

from 0.29.31 to HEAD

### Additional context

Maybe need to extend #4764 to more cases? I think this might have to do with the `CYTHON_PEP489_MULTI_PHASE_INIT=0` for PyPy. If I define it to `1` the warning goes away and NumPy tests pass. I couldn't easily grep for an existing test for importing another module within a module, but there must be one.

### PR DESCRIPTION
**Pull request by mattip**
Closes #5413 

I think the warning will stil appear in codepaths that do not define `CYTHON_PEP489_MULTI_PHASE_INIT=1`, like in `CYTHON_LIMITED_API`. But this fixes the PyPy case.

### CODE REVIEW

**Comment by mattip** [comment]
The [pypy 3.9](https://github.com/cython/cython/actions/runs/4849901110/jobs/8642324499?pr=5414) CI run was successful, and I also confirmed this works in NumPy.

**Comment by mattip** [comment]
CI was ~22 minutes vs [17.5 minutes here](https://github.com/cython/cython/actions/runs/4844580623/jobs/8632940460) and [30 minutes here](https://github.com/cython/cython/actions/runs/4843721373/jobs/8631537852) so it is hard to say anything conclusive about performance.

**Comment by da-woods** [comment]
> hard to say anything conclusive about performance.

Unless you know anything very PyPy specific, I'd be surprised if this affected performance. I think the time is largely dominated by C compile time and I don't think multi-phase init hugely changes that. I think it's mostly about better compatibility (e.g. knowing `__file__` when evaluating the module scope, and things like that)

File: Cython/Utility/ModuleSetupCode.c [hunk 1]
```diff
@@ -159,7 +159,7 @@
     #define CYTHON_PEP487_INIT_SUBCLASS (PY_MAJOR_VERSION >= 3)
   #endif
   #undef CYTHON_PEP489_MULTI_PHASE_INIT
-  #define CYTHON_PEP489_MULTI_PHASE_INIT 0
+  #define CYTHON_PEP489_MULTI_PHASE_INIT 1
```
Comment (scoder) [inline Cython/Utility/ModuleSetupCode.c:162]: Could we be specific about the minimum PyPy version here that supports it?

**Comment by scoder** [comment]
Thanks

**Comment by scoder** [comment]
Backported to 0.29.x.

**Comment by mattip** [comment]
Cool, thanks.

**Pull request merged**